slack.ts

```diff
@@ -0,0 +1,69 @@
+import * as querystring from "querystring";
+import fetch from "node-fetch";
+
+const tkn = process.env.SLACK_TOKEN;
```

Comment:
These should be checked to not be undefined, a good way to do this is to wrap all the functions in a class and check in the constructor like so: https://github.com/HackGT/buzzer/blob/master/server/plugins/Twitter.ts#L32.